nemo/collections/llm/tools/auto_configurator/core/search_config.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Optional
+
+from nemo.collections.llm.tools.auto_configurator.core.training_config import search_training_config
+from nemo.collections.llm.tools.auto_configurator.core.utils import generic_base_config
+
+SUPPORTED_MODELS = [
+    "gpt3",
+    "llama",
+    "mixtral",
+    "mistral",
+    "gemma",
+]
+
+
+def search_configs(cfg: dict):
```

Comment:
This function is hard to use since it takes in a free-form dictionary. Could we just introduce a dataclass instead? The dataclass could contain docs as well